tests/gamma/csm_observability_test.py

```diff
@@ -45,39 +100,389 @@ def setUpClass(cls):
+                    ):
+                        # the remote workload name for the client is the name
+                        # we set on the server
+                        self.assertEqual(_CSM_WORKLOAD_NAME_SERVER, label_value)
+                    elif (
+                        label_key == "csm_remote_workload_name"
+                        and self._is_server_metric(metric)
+                    ):
+                        # the remote workload name for the server is the name
+                        # we set on the client
+                        self.assertEqual(_CSM_WORKLOAD_NAME_CLIENT, label_value)
+                    elif (
+                        label_key == "csm_remote_workload_namespace_name"
+                        and self._is_client_metric(metric)
+                    ):
+                        # the server namespace name
+                        self.assertEqual(self.server_namespace, label_value)
+                    elif (
+                        label_key == "csm_remote_workload_namespace_name"
+                        and self._is_server_metric(metric)
+                    ):
+                        # the client namespace name
+                        self.assertEqual(self.client_namespace, label_value)
+                    elif label_key == "csm_remote_workload_project_id":
+                        self.assertEqual(self.project, label_value)
+                    elif label_key == "csm_remote_workload_type":
+                        # a hardcoded value
+                        self.assertEqual("gcp_kubernetes_engine", label_value)
+                    elif label_key == "csm_service_name":
+                        # the service name
+                        self.assertEqual(
+                            self.server_runner.service_name, label_value
+                        )
+                    elif label_key == "csm_service_namespace_name":
+                        # the server namespace name
+                        self.assertEqual(self.server_namespace, label_value)
+                    elif (
+                        label_key == "csm_workload_canonical_service"
+                        and self._is_client_metric(metric)
+                    ):
+                        # the client workload canonical service name
+                        self.assertEqual(
+                            _CSM_CANONICAL_SERVICE_NAME_CLIENT, label_value
+                        )
+                    elif (
+                        label_key == "csm_workload_canonical_service"
+                        and self._is_server_metric(metric)
+                    ):
+                        # the server workload canonical service name
+                        self.assertEqual(
+                            _CSM_CANONICAL_SERVICE_NAME_SERVER, label_value
+                        )
+
+        # Testing the values of monitored resource labels
+        with self.subTest("9_check_resource_label_values"):
```

Comment:
i'm fine with the checks on the resource labels but again i don't care about them

Reply:
Same as above.